### PR TITLE
Move 2FA registration to IBootstrap

### PR DESCRIPTION
**Pull request by rullzer**
1 Bootstrap to rule them all - _Ash Bootstrapzg durbatulûk,_
1 place to find them - _ash plazg gimbatul,_
1 place to code it all - _ash plazg codatulûk,_
And in darkness bind to them - _agh burzum-ishi krimpatul_

### CODE REVIEW

File: lib/private/AppFramework/Bootstrap/RegistrationContext.php [hunk 1]
```diff
@@ -92,7 +92,10 @@ class RegistrationContext {
 	private $templateProviders = [];
 
 	/** @var ServiceRegistration<INotifier>[] */
-	private $notifierServices;
+	private $notifierServices = [];
+
+	/** @var ServiceRegistration<\OCP\Authentication\TwoFactorAuth\IProvider> */
```
Comment (ChristophWurst) [inline lib/private/AppFramework/Bootstrap/RegistrationContext.php:97]:
```suggestion
	/** @var ServiceRegistration<\OCP\Authentication\TwoFactorAuth\IProvider>[] */
```

File: lib/public/AppFramework/Bootstrap/IRegistrationContext.php [hunk 2]
```diff
@@ -219,4 +220,13 @@ public function registerTemplateProvider(string $providerClass): void;
 	 * @since 22.0.0
 	 */
 	public function registerNotifierService(string $notifierClass): void;
+
+	/**
+	 * Register an Two Factor Provider
```
Comment (ChristophWurst) [inline lib/public/AppFramework/Bootstrap/IRegistrationContext.php:225]:
```suggestion
	 * Register a two-factor provider
```

**Review by ChristophWurst — CHANGES_REQUESTED** [review]

File: lib/private/AppFramework/Bootstrap/RegistrationContext.php [hunk 3]
```diff
@@ -479,4 +493,11 @@ public function getTemplateProviders(): array {
 	public function getNotifierServices(): array {
 		return $this->notifierServices;
 	}
+
+	/**
+	 * @return ServiceRegistration<\OCP\Authentication\TwoFactorAuth\IProvider>[]
+	 */
+	public function getTwoFactorProvider(): array {
```
Comment (ChristophWurst) [inline lib/private/AppFramework/Bootstrap/RegistrationContext.php:500]: nitpick

```suggestion
	public function getTwoFactorProviders(): array {
```

**Review by ChristophWurst — APPROVED** [review]

**Review by MorrisJobke — APPROVED** [review]
Looks good 👍

**Pull request merged**